### ISSUE TITLE
walk_files's prefix=False parameter does not make sense

### ISSUE THREAD

**Issue by mthrok** [pytorch/audio#815]
When there are files spread across nested directories, giving `prefix=False` would return file names without intermediate directory structures. The caller of the `walk_files` know the path to the root directory and receives filenames without intermediate files. This is not useful.

## Example

```
$ mkdir tmp
$ mkdir tmp/d1
$ mkdir tmp/d2
$ mkdir tmp/d3
$ touch tmp/f1.txt
$ touch tmp/f2.txt
$ touch tmp/f3.txt
$ touch tmp/d1/f4.txt
$ touch tmp/d1/f5.txt
$ touch tmp/d1/f6.txt
$ mkdir tmp/d1/d2/
$ touch tmp/d1/d2/f7.txt
$ touch tmp/d1/d2/f8.txt
$ touch tmp/d1/d2/f9.txt
$ tree
tmp/
|-- d1
|   |-- d2
|   |   |-- f7.txt
|   |   |-- f8.txt
|   |   `-- f9.txt
|   |-- f4.txt
|   |-- f5.txt
|   `-- f6.txt
|-- d2
|-- d3
|-- f1.txt
|-- f2.txt
`-- f3.txt

4 directories, 9 files
```

```
$ python -c 'import torchaudio;print(list(torchaudio.datasets.utils.walk_files("tmp", "txt", prefix=False)))'
['f1.txt', 'f2.txt', 'f3.txt', 'f4.txt', 'f5.txt', 'f6.txt', 'f7.txt', 'f8.txt', 'f9.txt']
```

What users can do with returned value of `'f9.txt'`??

## Proposal

Change the parameter to `relative=False/True` so that `walk_files` function returns paths relative to the given root directory or full path.

**Comment by mthrok** [comment]
oops I missed `prefix=True` parameter.

**Comment by mthrok** [comment]
I looked into this and most of the dataset implementation uses `prefix=False`, this means they expect single level directory structure, so there is no need to use `os.walk`. `os.listdir` suffices.

Only `speechcommands` uses `prefix=True`, but this one seemingly requires only intermediate path.

```
$ grep 'walk_files(' -A 3 -r torchaudio
torchaudio/datasets/vctk.py:        walker = walk_files(
torchaudio/datasets/vctk.py-            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
torchaudio/datasets/vctk.py-        )
torchaudio/datasets/vctk.py-        walker = filter(lambda w: self._except_folder not in w, walker)
--
torchaudio/datasets/speechcommands.py:        walker = walk_files(self._path, suffix=".wav", prefix=True)
torchaudio/datasets/speechcommands.py-        walker = filter(lambda w: HASH_DIVIDER in w and EXCEPT_FOLDER not in w, walker)
torchaudio/datasets/speechcommands.py-        self._walker = list(walker)
torchaudio/datasets/speechcommands.py-
--
torchaudio/datasets/utils.py:def walk_files(root: str,
torchaudio/datasets/utils.py-               suffix: Union[str, Tuple[str]],
torchaudio/datasets/utils.py-               prefix: bool = False,
torchaudio/datasets/utils.py-               remove_suffix: bool = False) -> Iterable[str]:
--
torchaudio/datasets/librispeech.py:        walker = walk_files(
torchaudio/datasets/librispeech.py-            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
torchaudio/datasets/librispeech.py-        )
torchaudio/datasets/librispeech.py-        self._walker = list(walker)
--
torchaudio/datasets/yesno.py:        walker = walk_files(
torchaudio/datasets/yesno.py-            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
torchaudio/datasets/yesno.py-        )
torchaudio/datasets/yesno.py-        self._walker = list(walker)
--
torchaudio/datasets/libritts.py:        walker = walk_files(
torchaudio/datasets/libritts.py-            self._path, suffix=self._ext_audio, prefix=False, remove_suffix=True
torchaudio/datasets/libritts.py-        )
torchaudio/datasets/libritts.py-        self._walker = list(walker)
```

**Comment by vincentqb** [comment]
> Change the parameter to `relative=False/True` so that `walk_files` function returns paths relative to the given root directory or full path.

LGTM.

**Comment by mthrok** [comment]
Resolved by #1109
